### PR TITLE
ctlcmd/tests: tests tweaks (followup to #6322)

### PR DESCRIPTION
**Pull request by stolowski**
Small tweaks/clarification as a followup to #6322 (thanks @mvo5!).

### CODE REVIEW

File: tests/lib/snaps/snapctl-hooks/meta/hooks/configure [hunk 1]
```diff
@@ -67,7 +67,9 @@ test_get_nested() {
       exit 1
   fi
   expected_output='{\n\t"key1": "a",\n\t"key2": "b"\n}'
-  if ! ( echo "$output" | grep -z -q -P "$expected_output" ); then
+  # note: "echo" is a built-in of sh and doesn't support -e flag, use /bin/echo.
```
Comment (zyga) [inline tests/lib/snaps/snapctl-hooks/meta/hooks/configure]: We can use `command echo` to execute the actual external program, not the shell built-in.

Comment (stolowski) [reply tests/lib/snaps/snapctl-hooks/meta/hooks/configure]: hmm, actually.. that doesn't work in sh, a quick test:

```
$ sh
$ command echo -e "foo"
-e foo
```
I'm inclined to leave the above as is after spending too much time on it in previous PR already ;)

File: overlord/hookstate/ctlcmd/get_test.go [hunk 2]
```diff
@@ -209,9 +209,10 @@ func (s *getSuite) TestGetPartialNestedStruct(c *C) {
 		c.Assert(string(stderr), Equals, "")
 		c.Check(string(stdout), Equals, test.stdout)
 
+		// transaction not commited, drop it
```
Comment (pedronis) [inline overlord/hookstate/ctlcmd/get_test.go]: typo, committed

File: tests/lib/snaps/snapctl-hooks/meta/hooks/configure (hunk 1 above)
Comment (mvo5) [inline tests/lib/snaps/snapctl-hooks/meta/hooks/configure]: This should work in spread fwiw, it uses "bash" as its shell.

Comment (stolowski) [reply tests/lib/snaps/snapctl-hooks/meta/hooks/configure]: It didn't; it's a hook, so it runs with whatever core image has, I didn't investigate further.

**Pull request merged**